src/server/database/mongodb-adaptor.js

```diff
@@ -8,7 +8,9 @@ class MongoCollectionAdaptor {
   }
 
   put (doc) {
-    return this.collection.insertOne(doc)
+    return new Promise((resolve, reject) => {
+      this.collection.insertOne(doc).then(() => resolve(doc)).catch((err) => reject(err))
+    })
```

Comment:
It returns the inserted document's _id. Continuing with the document stored in this instance of the app might lead to stale data (if someone updates the document while you have it in memory). Will you check for those kind of conflicts?